[Core deprecations] Add title field to deprecation info (#109840)
elastic/kibana: src/core/server/deprecations/README.mdx

        return completeConfig;
      }
      addDeprecation({
        title: i18n.translate('kibana_legacy.deprecations.defaultAppIdTitle', {
          defaultMessage: 'Setting "kibana.defaultAppId" is deprecated',
        }),
        message: i18n.translate('kibana_legacy.deprecations.defaultAppIdMessage', {

>           defaultMessage: '"kibana.defaultAppId" is deprecated and will be removed in 8.0. Please use the "defaultRoute" advanced setting instead.',

Reviewer: ```suggestion
          defaultMessage: 'Use the "defaultRoute" advanced setting instead. "kibana.defaultAppId" will be removed in 8.0. ',
```

        }),
        correctiveActions: {
          manualSteps: [
            i18n.translate('kibana_legacy.deprecations.defaultAppId.manualStepOneMessage', {
              defaultMessage: 'Go to Stack Management > Advanced Settings',
            }),
            i18n.translate('kibana_legacy.deprecations.defaultAppId.manualStepTwoMessage', {

>               defaultMessage: 'Update the "defaultRoute" setting under the General section',

Reviewer: ```suggestion
              defaultMessage: 'Update the defaultRoute setting in the General section',
```

            }),
            i18n.translate('kibana_legacy.deprecations.defaultAppId.manualStepThreeMessage', {

>               defaultMessage: 'Remove "kibana.defaultAppId" from the kibana.yml config file',

Reviewer: ```suggestion
              defaultMessage: 'Remove kibana.defaultAppId from the kibana.yml config file.',
```

### Example
```ts
import { DeprecationsDetails, GetDeprecationsContext } from 'src/core/server';
import { i18n } from '@kbn/i18n';

async function getDeprecations({ esClient, savedObjectsClient }: GetDeprecationsContext): Promise<DeprecationsDetails[]> {
  const deprecations: DeprecationsDetails[] = [];
  const testDashboardUser = await getTestDashboardUser(savedObjectsClient);

  if (testDashboardUser) {
  deprecations.push({
    title: i18n.translate('security.deprecations.kibanaUserRoleTitle', {

>       defaultMessage: 'Found users using deprecated roles',

Reviewer: Deprecated roles are assigned to some users

          }
      },
      manualSteps: [
        i18n.translate('security.deprecations.kibanaUserRole.manualStepOneMessage', {

>           defaultMessage: 'Using Kibana user management, change all users using the kibana_user role to the "kibana_admin" role.',

Reviewer: ```suggestion
          defaultMessage: 'Switch all users with the kibana_user role to the kibana_admin role in Management > Security > Users.',
```

        }),
        i18n.translate('security.deprecations.kibanaUserRole.manualStepTwoMessage', {

>           defaultMessage: 'Using Kibana role-mapping management, change all role-mappings which assing the "kibana_user" role to the "kibana_admin" role.'
>         }),

Reviewer: ```suggestion
          defaultMessage: 'Update all mappings in Management > Security > Role Mappings to assign the kibana_admin role instead of the kibana_user role.'
        }),
```
